### ISSUE TITLE
Secret notes

### ISSUE THREAD

**Issue by waba4mume** [MUME/MMapper#32]
Use case: There is some interesting loot to be found hidden around Arda, for explorers who bother reading the descs. I do not want to make that information public when I export a base (no secrets) map.

Suggested course of action: Add a *secret notes* field that is not exported by the *Export base map* feature. The field should be searchable. Or some more complex scheme that would allow encoding that information in room flags & normal notes while still not exporting them.

This can probably be bundled with #29.

**Comment by teoli2003** [comment]
Yes, this is one of the reason that I don't share my personal map (except with 3 other players, 2 of them not playing regularly). (Nope, it is not for the about 10 rooms not published on public maps :-) )

Improving the export base map feature is likely the correct way to go. Now that we have types of infomarks (locality, mob name, object, herbs, comments, …), we should be able to have some granularity there (secret exits are always tricky to hide as there are rooms behind them ;-) ).

We could add a 'hidden' flag on infomarks (for any type and class) as well as for room notes (1 note for 1 room currently).
Alternative: a new class of infomark (hidden) and considering each room note (not displayed) as 'hidden'.
